engine/src/main/java/com/datatorrent/stram/webapp/OperatorDiscoverer.java

```diff
@@ -565,13 +581,16 @@ private JSONArray enrichProperties(String operatorClass, JSONArray properties) t
         result.put(propJ);
         continue;
       }
-      if (oci.invisibleGetSetMethods.contains(getPrefix + propName) || oci.invisibleGetSetMethods.contains(setPrefix + propName)) {
+      MethodInfo setterInfo = oci.setMethods.get(setPrefix + propName);
+      MethodInfo getterInfo = oci.getMethods.get(getPrefix + propName);
+
+      if ((getterInfo != null && getterInfo.omitFromUI) || (setterInfo != null && setterInfo.omitFromUI)) {
         continue;
       }
-      MethodInfo methodInfo = oci.setMethods.get(setPrefix + propName);
-      methodInfo = methodInfo == null ? oci.getMethods.get(getPrefix + propName) : methodInfo;
-      if (methodInfo != null) {
-        addTagsToProperties(methodInfo, propJ);
+      if (setterInfo != null) {
```

Comment:
both setterInfo and getterInfo can be null. With ObjectUtils.firstNotNull I have to capture the result in a variable and then check if that is not null to perform this operation.
I think these utils make the code compact only when there is at least one non-null argument.

So I don't think it will make the code compact. Don't mind changing this but there is not much gained here